funsor/tensor.py

```diff
@@ -772,23 +767,27 @@ def eager_reshape_tensor(op, arg):
     return Tensor(data, arg.inputs, arg.dtype)
 
 
-@eager.register(Unary, ops.SumOp, Tensor)
-def eager_sum_tensor(op, arg):
+@eager.register(Unary, ops.ReductionOp, Tensor)
+def eager_reduction_tensor(op, arg):
+    if not arg.output.shape:
+        return arg
```

Comment:
This won't work for `argmin`,`argmax`.  Are those two really `ReductionOp`s?  Maybe we should call them `ArgreductionOp`s or sth?